contracts/VeRBNRewards.sol

```diff
@@ -0,0 +1,247 @@
+        return _earnedReward(account);
+    }
+
+    /** @notice use to update rewards on veRBN balance changes.
+        @dev called by veRBN
+     *  @return true
+     */
+    function updateReward(address _account)
+        external
+        _updateReward(_account)
+        returns (bool)
+    {
+        require(msg.sender == address(veToken), "!authorized");
+
+        return true;
+    }
+
+    /**
+     * @notice
+     *  Get rewards for an account
+     * @dev rewards are transfer to _account
+     * @param account to claim rewards for
+     * @return true
+     */
+    function getRewardFor(address account) external returns (bool) {
+        _getReward(account, false);
+        return true;
+    }
+
+    /**
+     * @notice
+     *  Get rewards
+     * @param _lock should it lock rewards into veRBN
+     * @return true
+     */
+    function getReward(bool _lock) external returns (bool) {
+        _getReward(msg.sender, _lock);
+        return true;
+    }
+
+    /**
+     * @notice
+     *  Get rewards
+     * @return true
+     */
+    function getReward() external returns (bool) {
+        _getReward(msg.sender, false);
+        return true;
+    }
+
+    function _getReward(address _account, bool _lock)
+        internal
+        _updateReward(_account)
+    {
+        uint256 reward = rewards[_account];
+        rewards[_account] = 0;
+        if (reward == 0) return;
+
+        if (_lock) {
+            SafeERC20.safeApprove(rewardToken, address(veToken), reward);
+            veToken.deposit_for(msg.sender, reward);
+        } else {
+            SafeERC20.safeTransfer(rewardToken, _account, reward);
+        }
+
+        emit RewardPaid(_account, reward);
+    }
+
+    /**
+     * @notice
+     *  Donate tokens to distribute as rewards
+     * @dev Do not trigger rewardRate recalculation
+     * @param _amount token to donate
+     * @return true
+     */
+    function donate(uint256 _amount) external returns (bool) {
+        require(_amount != 0, "==0");
+        IERC20(rewardToken).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _amount
+        );
+        queuedRewards = queuedRewards + _amount;
+        return true;
+    }
+
+    /**
+     * @notice
+     * Add new rewards to be distributed over a week
+     * @dev Trigger rewardRate recalculation using _amount and queuedRewards
+     * @param _amount token to add to rewards
+     * @return true
+     */
+    function queueNewRewards(uint256 _amount) external returns (bool) {
+        require(_amount != 0, "==0");
+        IERC20(rewardToken).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _amount
+        );
+
+        _amount = _amount + queuedRewards;
+        _notifyRewardAmount(_amount);
+        queuedRewards = 0;
+
+        return true;
+    }
+
+    function _notifyRewardAmount(uint256 reward)
+        internal
+        _updateReward(address(0))
```

Comment:
Also a bit confused by this function name.  `notify` implies no side effects

Reply:
also synthetix